ArduPlane/Parameters.cpp

```diff
@@ -522,13 +522,30 @@ const AP_Param::Info Plane::var_info[] = {
+    // @Description: Enable ground control station telemetry failsafe. Failsafe will trigger after FS_GCS_TIMEOUT seconds of no MAVLink heartbeat messages. There are three possible enabled settings. Seeing FS_GCS_ENABL to 1 means that GCS failsafe will be triggered when the aircraft has not received a MAVLink HEARTBEAT message. Setting FS_GCS_ENABL to 2 means that GCS failsafe will be triggered on either a loss of HEARTBEAT messages, or a RADIO_STATUS message from a MAVLink enabled 3DR radio indicating that the ground station is not receiving status updates from the aircraft, which is indicated by the RADIO_STATUS.remrssi field being zero (this may happen if you have a one way link due to asymmetric noise on the ground station and aircraft radios).Setting FS_GCS_ENABL to 3 means that GCS failsafe will be triggered by Heartbeat(like option one), but only in AUTO mode. WARNING: Enabling this option opens up the possibility of your plane going into failsafe mode and running the motor on the ground it it loses contact with your ground station. If this option is enabled on an electric plane then you should enable ARMING_REQUIRED.
     // @Values: 0:Disabled,1:Heartbeat,2:HeartbeatAndREMRSSI,3:HeartbeatAndAUTO
     // @User: Standard
     GSCALAR(gcs_heartbeat_fs_enabled, "FS_GCS_ENABL", GCS_FAILSAFE_OFF),
 
+    // @Param: FS_GCS_ACTION
+    // @DisplayName: GCS failsafe action
+    // @Description: The action to take on a GCS failsafe event. If the aircraft was in a stabilization or manual mode when failsafe started and a GCS failsafe occurs then it will change to RTL mode if FS_GCS_ACTION is 0 or 1, and will change to FBWA if FS_GCS_ACTION is set to 2. If the aircraft was in an auto mode (such as AUTO or GUIDED) when the failsafe started then it will continue in the auto mode if FS_GCS_ACTION is set to 0, will change to RTL mode if FS_GCS_ACTION is set to 1 and will change to FBWA mode if FS_GCS_ACTION is set to 2. If FS_GCS_ACTION is set to 3, the parachute will be deployed (make sure the chute is configured and enabled). 
+    // @Values: 0:Continue,1:ReturnToLaunch,2:Glide,3:Deploy Parachute
+    // @User: Standard
+    GSCALAR(fs_action_gcs, "FS_GCS_ACTION", FS_ACTION_GCS_CONTINUE),
```

Comment:
Would you be ameniable to using the FailsafeAction enum here and adding any other cases we don't currently have to the enum/handler for it? I'd like to see us use a more common and flexible set of actions for failsafes moving forwards.

Reply:
Not sure I understand you. Are you talking about the FaiisafeAction enum in @peterbarker 's code?  If so, didn't you just say that's a third rail?  Of course, you could be speaking about another FailsafeAction enum.

Reply:
@mday299 Sorry I should have been a lot clearer. I meant specifically the actions in this enum: https://github.com/ArduPilot/ardupilot/blob/master/ArduPlane/Plane.h#L1034 currently they are used in the battery failsafe, but all of the listed actions I believe are quite valid for GCS failsafes, and any useful new ones would be probably be good for battery failsafe actions.

The reason this part isn't as bad to touch is because the list is unique per vehicle type at the moment, but external libraries can still interact with it.